content/en/docs/reference/using-api/api-concepts.md

```diff
@@ -340,48 +374,70 @@ Clients that receive a response in `application/vnd.kubernetes.protobuf` that do
 
 Resources are deleted in two phases: 1) finalization, and 2) removal.
 
-```go
+
+Here is an excerpt of an object with a finalizer set, prior to deletion:
+```yaml
 {
   "kind": "ConfigMap",
   "apiVersion": "v1",
+  …
   "metadata": {
     "finalizers": {"url.io/neat-finalization", "other-url.io/my-finalizer"},
     "deletionTimestamp": nil,
   }
 }
 ```
 
-When a client first deletes a resource, the `.metadata.deletionTimestamp` is set to the current time.
-Once the `.metadata.deletionTimestamp` is set, external controllers that act on finalizers
-may start performing their cleanup work at any time, in any order.
-Order is NOT enforced because it introduces significant risk of stuck `.metadata.finalizers`.
-`.metadata.finalizers` is a shared field, any actor with permission can reorder it.
-If the finalizer list is processed in order, then this can lead to a situation
-in which the component responsible for the first finalizer in the list is
+To delete a resource, you send a `DELETE` HTTP request to the API server. If the
+object has any finalizer configured, the API server does not immediately delete the
+resource. Instead, the API server sets its `.metadata.deletionTimestamp` to the current
+time.
+
+Once the `.metadata.deletionTimestamp` is set,
+{{< glossary_tooltip term_id="controller" text="controllers">}} that act on the object's
+finalizers may start performing their cleanup. The controllers can perform finalization
+at any time and in any order.
+
+{{< note >}}
+Enforced ordering would introduce significant risk of finalization deadlock.
+
+`.metadata.finalizers` is a shared field; any actor with permission to write that
+field can reorder it. If the finalizer list is processed in order, this can lead to
+a situation in which the component responsible for the first finalizer in the list is
 waiting for a signal (field value, external system, or other) produced by a
-component responsible for a finalizer later in the list, resulting in a deadlock.
-Without enforced ordering finalizers are free to order amongst themselves and
+component responsible for a finalizer later in the list, and the finalization is
+unable to proceed.
+
+Without enforced ordering, finalizers are free to order amongst themselves and
 are not vulnerable to ordering changes in the list.
+{{< /note >}}
 
-Once the last finalizer is removed, the resource is actually removed from etcd.
+After a finalizer completes its tasks, the relevant controller marks that completion
+by removing its own entry from the list of finalizers on that resource.
 
+Once the last finalizer is removed, the resource is actually removed from etcd and,
```

Comment:
nit: remove "actually"

Reply:
I've omitted this change from the latest revision.